packages/fxa-settings/src/lib/flow-event.ts

```diff
@@ -0,0 +1,86 @@
+import sentryMetrics from '@fxa-shared/lib/sentry';
+
+interface FlowEventParams {
+  device_id?: string;
+  flow_id?: string;
+  flow_begin_time?: number;
+}
+
+interface FlowEventData {
+  deviceId: string;
+  flowBeginTime: number;
+  flowId: string;
+}
+
+let initialized = false;
+let flowEventData: FlowEventData;
+
+function shouldSend() {
+  return initialized && window.navigator.sendBeacon;
+}
+
+function postMetrics(eventData: object) {
+  // This is not an Action insofar that it has no bearing on the app state.
+  window.navigator.sendBeacon('/metrics', JSON.stringify(eventData));
+}
+
+export function init(eventData: FlowEventParams, redirectIfMissing = false) {
+  if (!initialized) {
+    if (eventData.device_id && eventData.flow_begin_time && eventData.flow_id) {
+      flowEventData = {
+        deviceId: eventData.device_id,
+        flowBeginTime: eventData.flow_begin_time,
+        flowId: eventData.flow_id,
+      };
+      initialized = true;
+    } else if (redirectIfMissing) {
+      let redirectPath = window.location.pathname;
+      if (window.location.search) {
+        redirectPath += window.location.search;
+      }
+
+      return window.location.replace(
+        `${window.location.origin}/get_flow?redirect_to=${encodeURIComponent(
+          redirectPath
+        )}`
+      );
+    }
+  }
+}
+
+export function logAmplitudeEvent(
```

Comment:
I need some time tomorrow to look at this thoroughly.  This stuff looks copied from payments, _however_, it is now sending to content server.  The metrics endpoints on content and payments are not identical.

Reply:
Good point. I did some checking, and the resulting `eventData` object in `flow-events.ts` is the bare minimum structure needed to successfully deliver a metrics payload to the content server. I've only set up the values to get it to send; they still need to be properly set and validated, but if it's cool with you I don't think that should block this PR as we have a [few other issues](https://jira.mozilla.com/browse/FXA-1692) that should cover it.